### PR TITLE
BE 20_BE 38_BE_49_add interviewer entity_tests

### PR DESCRIPTION
**Pull request by ValentinKhorunzhyn**

### CODE REVIEW

**Review by serkushner — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/project/internship/controller/InterviewerController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    private static final String ID = "/{id}";
+
+    @GetMapping(value = ID)
+    public ResponseEntity<InterviewerDto> getInterviewerById(@PathVariable Long id) throws EntityNotFoundException {
+        return ResponseEntity.ok(interviewerService.getById(id));
+    }
+
+    @GetMapping
+    public ResponseEntity<List<InterviewerDto>> findAllInterviewers(@RequestParam(value = "search", required = false) String search,
```
[the remark this answers is not in the document]
Comment (ValentinKhorunzhyn) [reply src/main/java/com/exadel/project/internship/controller/InterviewerController.java]: Method renamed

File: src/main/java/com/exadel/project/internship/service/InterviewerService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    private void fillInterviewer(InterviewerDto interviewerDto, Interviewer interviewer) {
+        interviewer.setName(interviewerDto.getName());
+        interviewer.setSurname(interviewerDto.getSurname());
+        interviewer.setEmail(interviewerDto.getEmail());
+        interviewer.setPhone(interviewerDto.getPhone());
+        interviewer.setType(interviewerDto.getType());
+        interviewer.setSkype(interviewerDto.getSkype());
+    }
```
[the remark this answers is not in the document]
Comment (ValentinKhorunzhyn) [reply src/main/java/com/exadel/project/internship/service/InterviewerService.java]: Moved into update method and wrapped by one records

File: src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    @Test
+    void getInterviewerById() throws Exception {
+        InterviewerDto dto = new InterviewerDto();
+        dto.setId(1L);
+        dto.setName("firstName");
+        dto.setSurname("lastName");
+        dto.setEmail("test@mail.ru");
+        dto.setPhone("80556958574");
+        dto.setType(InterviewerType.TECH);
+        dto.setSkype("testSkype");
```
[the remark this answers is not in the document]
Comment (ValentinKhorunzhyn) [reply src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java]: InterviewerTestData created with two methods which will return HR ore TECH Interviewer

File: src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class InterviewerRestControllerTests {
```
[the remark this answers is not in the document]
Comment (ValentinKhorunzhyn) [reply src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java]: Renamed

File: src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        InterviewerDto dto = new InterviewerDto();
+        dto.setId(1L);
+        dto.setName("firstName");
+        dto.setSurname("lastName");
+        dto.setEmail("test@mail.ru");
+        dto.setPhone("80556958574");
+        dto.setType(InterviewerType.TECH);
+        dto.setSkype("testSkype");
+
+        doReturn(dto).when(interviewerService).getById(1L);
```
[the remark this answers is not in the document]
Comment (ValentinKhorunzhyn) [reply src/test/java/com/exadel/project/controller/InterviewerRestControllerTests.java]: InterviewerCrudTest include tests with mock (interviewerRepository)
JsonSerialization TEst and Service tests are deleted.

**Review by serkushner — APPROVED** [review]

**Review by nidaffs — APPROVED** [review]

**Pull request merged**